presto-hive/src/main/java/io/prestosql/plugin/hive/util/Statistics.java

```diff
@@ -83,6 +83,10 @@ private Statistics() {}
 
     public static PartitionStatistics merge(PartitionStatistics first, PartitionStatistics second)
     {
+        if (first.getBasicStatistics().getRowCount().orElse(0L) == 0) {
```

Comment:
I do not like the fact that this is not symetric. The method signature does not require that table properties are passed as first argument. Can you make the condition symmetric or at rename the argument to add meaning to parameters?

Reply:
Good point. Also, I am not sure i like the `.orElse` part, since lack of information about row count it totally different from having information it's 0.
let me try.